### PR TITLE
Web user search

### PR DESCRIPTION
**Pull request by czue**
it came up recently that this would be a useful tool for FLOS to try and get more information about a user if they don't submit with a project space.

@proteusvacuum / anyone

image

### CODE REVIEW

File: corehq/apps/hqadmin/views.py [hunk 1]
```diff
@@ -561,6 +561,24 @@ def _lookup_id_in_couch(doc_id, db_name=None):
...
+    template = "hqadmin/web_user_lookup.html"
+    set_bootstrap_version3()
+    web_user_email = request.GET.get("q")
+    if not web_user_email:
+        return render(request, template, {})
+
+    web_user = WebUser.get_by_username(web_user_email)
+    if web_user is None:
+        messages.error(
+            request, "Sorry, no user found with email {}. Did you enter it correctly?".format(web_user_email)
```
Comment (proteusvacuum) [inline corehq/apps/hqadmin/views.py]: Should this be marked for translation? Or do we not do that for internal-only pages?
:fr:

File: corehq/apps/hqwebapp/models.py [hunk 2]
```diff
@@ -1536,6 +1536,8 @@ def sidebar_items(self):
                                 args=('pillow_errors',))},
                 {'title': _('Login as another user'),
                  'url': reverse(AuthenticateAs.urlname)},
+                {'title': _('Lookup user by email'),
```
Comment (proteusvacuum) [inline corehq/apps/hqwebapp/models.py]: nit: lookup is not a verb (should be look up)

**Comment by proteusvacuum** [comment]
:+1: on :ballot_box_with_check:

File: corehq/apps/hqadmin/views.py (hunk 1 above)
Comment (czue) [inline corehq/apps/hqadmin/views.py]: generally not a requirement. there's a tradeoff between "why does this code look wrong" and "creating a bunch of strings for the translators that don't actually matter". I personally go for the "less work for translators" route unless the surrounding code is already translated.

Comment (proteusvacuum) [inline corehq/apps/hqadmin/views.py]: Makes sense, thanks. 
I would also argue that this isn't necessarily an error condition, and we should show the fact that no users were found in the table (as we do when filters come up with no results), but for an internal facing page probably doesn't matter that much.

File: corehq/apps/hqwebapp/models.py (hunk 2 above)
Comment (czue) [inline corehq/apps/hqwebapp/models.py]: https://github.com/dimagi/commcare-hq/commit/05268a895c2b94373848279f7e3bf0c7efd42fae

File: corehq/apps/hqadmin/views.py (hunk 1 above)
Comment (czue) [inline corehq/apps/hqadmin/views.py]: yeah makes sense and also won't bother to change now

**Comment by czue** [comment]
:white_check_mark:

**Comment by proteusvacuum** [comment]
:boy: :mag: :mag_right:  :girl:

**Pull request merged**